### PR TITLE
Move updateInViewport to from resource to owners service impl

### PR DESCRIPTION
**Pull request by powerivq**
This PR moves updateInViewport to owners-impl.

1. Some unit tests have been disabled for other reasons so the fix cannot be validated.
2. There are references to methods in resources-impl from building-an-amp-extension.md. I think it would be better to update it once we finish rather than piece-by-piece.

### CODE REVIEW

**Comment by powerivq** [comment]
FYI e2e failure is due to the same ChromeDriver problem.

File: extensions/amp-carousel/0.1/test/test-scrollable-carousel.js [hunk 1]
```diff
@@ -347,7 +375,11 @@ describes.realWin(
           impl.goCallback(1, /*animate*/ false);
           impl.goCallback(-1, /*animate*/ false);
 
-          const updateInViewportSpy = sandbox.spy(impl, 'updateInViewport');
+          const ownerImpl = Services.ownersForDoc(impl.element);
+          const updateInViewportSpy = sandbox.spy(
```
Comment (lannka) [inline extensions/amp-carousel/0.1/test/test-scrollable-carousel.js]: move this to beforeEach to avoid code duplications

Comment (powerivq) [reply extensions/amp-carousel/0.1/test/test-scrollable-carousel.js]: The problem is that currently everything is constructed after beforeEach (https://github.com/ampproject/amphtml/blob/1264bc6d75844176af61d954c1428cc47b3805c1/extensions/amp-carousel/0.1/test/test-scrollable-carousel.js#L130).

We could refactor these tests so there wouldn't as many duplicates as we wish, but maybe it is better left on a separate PR.

Comment (lannka) [reply extensions/amp-carousel/0.1/test/test-scrollable-carousel.js]: the services installation is done in `describes.realWin`, owners service should be available already in beforeEach. Did you see it not working?

**Comment by powerivq** [comment]
@lannka Updated. It is mergable.

**Comment by lannka** [comment]
you can move the `spy` to beforeEach too.

**Comment by powerivq** [comment]
Again updated.

**Pull request merged**